subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/DependencyGraphBuilder.java

```diff
@@ -325,26 +327,45 @@ private void validateGraph(ResolveState resolveState) {
                 // We need to attach failures on unattached dependencies too, in case a node wasn't selected
                 // at all, but we still want to see an error message for it.
                 attachFailureToEdges(error, module.getUnattachedDependencies());
-            } else if (module.isVirtualPlatform() && module.hasCompetingForceSelectors()) {
+            } else if (module.isVirtualPlatform()) {
                 attachMultipleForceOnPlatformFailureToEdges(module);
             }
         }
     }
 
     private void attachMultipleForceOnPlatformFailureToEdges(ModuleResolveState module) {
-        List<EdgeState> forcedEdges = Lists.newArrayList();
+        List<EdgeState> forcedEdges = null;
+        boolean hasMultipleVersions = false;
+        String currentVersion = module.maybeFindForcedPlatformVersion();
         Set<ModuleResolveState> participatingModules = module.getPlatformState().getParticipatingModules();
         for (ModuleResolveState participatingModule : participatingModules) {
             for (EdgeState incomingEdge : participatingModule.getIncomingEdges()) {
-                if (incomingEdge.getSelector().isForce()) {
+                SelectorState selector = incomingEdge.getSelector();
+                if (selector.isForce() && !selector.isSoftForce()) {
                     // Filter out platform originating edges
```

Comment:
That comment should either be removed or move down above line 349

Reply:
Even better, the condition could be extracted to a method named `isPlatformOriginatingEdge` and the comment removed.